Export template function

GeorgeTaveras1231 opened the pull request:
This facilitates wrapping template behavior in other functions, and not needing to reference the tag function in many places.

In my case, I've created custom `Console` class that can be configured to have different prefixes, I'd like to be able to instantiate the class with a color template without needing to import `chalk` or `chalk-template` in multiple places

```js
const scopedConsole = new Console(/*Prefix*/'{blue SomePrefix}')
```

GeorgeTaveras1231 committed:
Export template function

This facilitates wrapping template behavior in other functions, and not needing to reference the tag function in many places.

sindresorhus commented:
I'm indifferent to exposing this. Deferring to @Qix-. However, this will need tests, docs, and TS types for it to be mergable.

Qix- commented:
Exposing it is fine, but agreed, needs tests, types and docs.

Docs need to mention that all input needs to be properly escaped - a lot of the "safety" comes from using tagged literals.

GeorgeTaveras1231 committed:
Add tests and types

Additionally add typescript as a dev dependency

GeorgeTaveras1231 committed:
Update README

GeorgeTaveras1231 commented:
@Qix- @sindresorhus I've updated the branch to include tests, types, and docs. I was a little unsure about what to do but welcome any feedback. I also added typescript as a dev dependency because it seemed that `npm test` requires typescript.

Qix- commented:
Thank you :) Published as `0.4.0`

sindresorhus commented on index.d.ts:
````diff
@@ -21,3 +21,15 @@ log(chalk.red.bgBlack(chalkTemplate`2 + 3 = {bold ${2 + 3}}`));
+```
+import { template } from 'chalk-template';
+
+log(template('Today is {red hot}')
````
Multiple typos

sindresorhus commented on index.d.ts:
````diff
@@ -21,3 +21,15 @@ log(chalk.red.bgBlack(chalkTemplate`2 + 3 = {bold ${2 + 3}}`));
+
+@example
+```
+import { template } from 'chalk-template';
````
Does not follow the existing code style.

sindresorhus commented on package.json:
```diff
@@ -49,6 +49,7 @@
 		"ava": "^3.15.0",
 		"cross-env": "^7.0.3",
 		"tsd": "^0.18.0",
+		"typescript": "^4.6.2",
```
Don't make unrelated changes.

sindresorhus commented on readme.md:
```diff
@@ -58,6 +58,16 @@ Note that function styles (`rgb()`, etc.) may not contain spaces between paramet
 All interpolated values (`` chalkTemplate`${foo}` ``) are converted to strings via the `.toString()` method. All curly braces (`{` and `}`) in interpolated value strings are escaped.
 
+## Template function
+
```
Readme is not in sync with index.d.ts

sindresorhus commented:
> Docs need to mention that all input needs to be properly escaped - a lot of the "safety" comes from using tagged literals.

This was not done.

sindresorhus commented:
I cleaned this up in https://github.com/chalk/chalk-template/commit/bf830d7b965c8c6144fbe0793ef4cc89480435d9.

Qix- commented:
Whoops sorry @sindresorhus, missed the mark on this one. Thanks for cleaning it up.
